### ISSUE TITLE
Option to disable escaping characters in autolink

### ISSUE THREAD

**Issue by jeffliu-LL** [jupyterlab/jupyterlab#8722]
### Problem
Currently, the autolinker automatically escapes html characters such as `&` and `,`. However, when building requests for certain REST services, these characters need to be unescaped in the url in order for them to be correct. This makes the printed url different from the actual url, and makes the autolinked url useless for testing the REST call.

For example, the following cell:
```python
import requests
wfs_url = "https://gis.ncdc.noaa.gov/arcgis/services/cdo/nexrad/MapServer/WFSServer?"
params = dict(service='WFS', request='GetCapabilities')
q = requests.Request('GET', wfs_url, params=params).prepare().url
print(q)
```
results in the following link being printed:
https://gis.ncdc.noaa.gov/arcgis/services/cdo/nexrad/MapServer/WFSServer?service=WFS&amp;request=GetCapabilities 
which is not a valid query.

whereas the actual text stored in the object `q` is:
https://gis.ncdc.noaa.gov/arcgis/services/cdo/nexrad/MapServer/WFSServer?service=WFS&request=GetFeature&typeName=nexrad:NEXRAD 
which is a valid query.

### Proposed Solution
Add an option to disable escaping characters in autolink, either for a single cell or globally, or an option to disable autolink altogether

**Comment by vovayartsev** [comment]
That would be extremely helpful, since `google_auth_oauthlib` lib relies on printing OAuth URL in a cell. Escaping breaks it: 

![image](https://user-images.githubusercontent.com/204231/91694227-594ed780-eb7d-11ea-9c82-c47a8341902a.png)

So examples from [pandas-gbq](https://pandas-gbq.readthedocs.io/en/latest/howto/authentication.html) work in classical **Jupyter Notebook** but not in **JupyterLab**

**Comment by jasongrout** [comment]
This is fixed for me in jlab 3 now:
<img width="924" alt="Screen Shot 2021-01-27 at 8 44 50 AM" src="https://user-images.githubusercontent.com/192614/106023742-f0ac4e00-607b-11eb-9d84-a2e57c3a0962.png">

I think it may have been fixed in 2.2.x as well.

**Comment by jasongrout** [comment]
Probably fixed in #8496.
